huaweicloud/services/nat/resource_huaweicloud_nat_snat_rule.go

```diff
@@ -0,0 +1,326 @@
+		NetworkId:    subnetId,
+		SourceType:   sourceType,
+	}
+
+	log.Printf("[DEBUG] The create options of the SNAT rule is: %#v", opts)
+	resp, err := snats.Create(client, opts)
+	if err != nil {
+		return diag.Errorf("error creating SNAT Rule: %s", err)
+	}
+	d.SetId(resp.ID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      snatRuleStateRefreshFunc(client, d.Id(), []string{"ACTIVE"}),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        3 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceSnatRuleRead(ctx, d, meta)
+}
+
+func resourceSnatRuleRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	natClient, err := cfg.NatGatewayClient(region)
+	if err != nil {
+		return diag.Errorf("error creating NAT v2 client: %s", err)
+	}
+
+	resp, err := snats.Get(natClient, d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "SNAT Rule")
+	}
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("gateway_id", resp.GatewayId),
+		d.Set("floating_ip_id", resp.FloatingIpId),
+		d.Set("floating_ip_address", resp.FloatingIpAddress),
+		d.Set("source_type", resp.SourceType),
+		d.Set("subnet_id", resp.NetworkId),
+		d.Set("cidr", resp.Cidr),
+		d.Set("status", resp.Status),
+		d.Set("description", resp.Description),
+	)
+	if err = mErr.ErrorOrNil(); err != nil {
+		return diag.Errorf("error saving SNAT rule fields: %s", err)
+	}
+	return nil
+}
+
+func resourceSnatRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	natClient, err := cfg.NatGatewayClient(region)
+	if err != nil {
+		return diag.Errorf("error creating NAT v2 client: %s", err)
+	}
+
+	ruleId := d.Id()
+	opts := snats.UpdateOpts{
+		GatewayId:   getNatGatewayId(d),
+		Description: utils.String(d.Get("description").(string)),
+	}
+	if d.HasChange("floating_ip_id") {
+		eipClient, err := cfg.NetworkingV1Client(region)
+		if err != nil {
+			return diag.Errorf("error creating VPC v1 client: %s", err)
+		}
+
+		eipIDs := d.Get("floating_ip_id").(string)
+		eipList := strings.Split(eipIDs, ",")
+		eipAddrs := make([]string, len(eipList))
+
+		// get EIP address from ID
+		for i, eipId := range eipList {
+			eIP, err := eips.Get(eipClient, eipId).Extract()
+			if err != nil {
+				return diag.Errorf("error fetching EIP (%s): %s", eipId, err)
+			}
+			eipAddrs[i] = eIP.PublicAddress
+		}
+
+		opts.FloatingIpAddress = strings.Join(eipAddrs, ",")
+	}
+
+	log.Printf("[DEBUG] The update options of the SNAT rule is: %#v", opts)
+	_, err = snats.Update(natClient, ruleId, opts)
+	if err != nil {
+		return diag.Errorf("error updating Snat rule: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error updating SNAT rule: %s", err)
```
keep consistent with line272

Reply:
done.